src/components/calculator.js

```diff
@@ -1,26 +1,191 @@
+        +/-
+      </button>
+      <button
+        className="operation-percent"
+        type="button"
+        onClick={() => {
+          handleClick('%');
+        }}
+      >
+        %
+      </button>
+      <button
+        className="operation-divition"
+        type="button"
+        onClick={() => {
+          handleClick('÷');
+        }}
+      >
+        ÷
+      </button>
+      <button
+        className="operation-multiplication"
+        type="button"
+        onClick={() => {
+          handleClick('x');
+        }}
+      >
+        x
+      </button>
+      <button
+        className="operation-minus"
+        type="button"
+        onClick={() => {
+          handleClick('-');
+        }}
+      >
+        -
+      </button>
+      <button
+        className="operation-add"
+        type="button"
+        onClick={() => {
+          handleClick('+');
+        }}
+      >
+        +
+      </button>
+      <button
+        className="operation-iqual"
+        type="button"
+        onClick={() => {
+          handleClick('=');
+        }}
+      >
+        =
+      </button>
+      <button
+        className="number9"
+        type="button"
+        onClick={() => {
+          handleClick('9');
+        }}
+      >
+        9
+      </button>
+      <button
+        className="number8"
+        type="button"
+        onClick={() => {
+          handleClick('8');
+        }}
+      >
+        8
+      </button>
+      <button
+        className="number7"
+        type="button"
+        onClick={() => {
+          handleClick('7');
+        }}
+      >
+        7
+      </button>
+      <button
+        className="number6"
+        type="button"
+        onClick={() => {
+          handleClick('6');
+        }}
+      >
+        6
+      </button>
+      <button
+        className="number5"
+        type="button"
+        onClick={() => {
+          handleClick('5');
+        }}
+      >
+        5
+      </button>
+      <button
+        className="number4"
+        type="button"
+        onClick={() => {
+          handleClick('4');
+        }}
+      >
+        4
+      </button>
+      <button
+        className="number3"
+        type="button"
+        onClick={() => {
+          handleClick('3');
+        }}
+      >
+        3
+      </button>
+      <button
+        className="number2"
+        type="button"
+        onClick={() => {
+          handleClick('2');
+        }}
+      >
+        2
+      </button>
+      <button
+        className="number1"
+        type="button"
+        onClick={() => {
+          handleClick('1');
+        }}
+      >
+        1
+      </button>
+      <button
+        className="number0"
+        type="button"
+        onClick={() => {
+          handleClick('0');
+        }}
+      >
+        0
+      </button>
+      <button
+        className="number-dot"
+        type="button"
+        onClick={() => {
+          handleClick('.');
+        }}
+      >
+        .
+      </button>
```

Comment:
[OPTIONAL] Well done on making the calculator work 🔢 but this piece of code can be improved 👨🏽‍💻 a lot of repetition is made and maybe you could consider using an array of objects. You can define an array of objects to represent the buttons in your calculator. Each object can contain the properties `name`, `className`, and `onClick`. Here's an example:
```
const buttons = [
  { name: 'AC', className: 'operationAC', onClick: () => handleClick('AC') },
  { name: '+/-', className: 'operation-change', onClick: () => handleClick('+/-') },
  { name: '%', className: 'operation-percent', onClick: () => handleClick('%') },
  { name: '÷', className: 'operation-divition', onClick: () => handleClick('÷') },
  { name: 'x', className: 'operation-multiplication', onClick: () => handleClick('x') },
  { name: '-', className: 'operation-minus', onClick: () => handleClick('-') },
  { name: '+', className: 'operation-add', onClick: () => handleClick('+') },
  { name: '=', className: 'operation-iqual', onClick: () => handleClick('=') },
  { name: '9', className: 'number9', onClick: () => handleClick('9') },
  { name: '8', className: 'number8', onClick: () => handleClick('8') },
  { name: '7', className: 'number7', onClick: () => handleClick('7') },
  { name: '6', className: 'number6', onClick: () => handleClick('6') },
  { name: '5', className: 'number5', onClick: () => handleClick('5') },
  { name: '4', className: 'number4', onClick: () => handleClick('4') },
  { name: '3', className: 'number3', onClick: () => handleClick('3') },
  { name: '2', className: 'number2', onClick: () => handleClick('2') },
  { name: '1', className: 'number1', onClick: () => handleClick('1') },
  { name: '0', className: 'number0', onClick: () => handleClick('0') },
  { name: '.', className: 'number-dot', onClick: () => handleClick('.') },
];
```
You can then use the `map` method to create the buttons in your `JSX`:
```
<section className="calculator-container">
  <div className="calculation-result">{result.next || result.total || '0'}</div>
  {buttons.map((button) => (
    <button key={button.name} className={button.className} type="button" onClick={button.onClick}>
      {button.name}
    </button>
  ))}
</section>
```
Note that we're using the key prop with the button name to help React efficiently update the UI when necessary.